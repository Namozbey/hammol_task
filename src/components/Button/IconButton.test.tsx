import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { IconButton } from ".";

describe("Testing IconButton", () => {
  it("Snapshots", () => {
    const { asFragment } = render(<IconButton />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("Fire click event", () => {
    const onClick = (e: React.MouseEvent<HTMLButtonElement>) => {
      expect(e).not.toBeUndefined();
      expect(e).not.toBeNull();
    };

    const { getByText } = render(
      <IconButton iconName="drop_down" onClick={onClick} />
    );
    const element = getByText("drop_down");
    fireEvent.click(element);
  });
});
