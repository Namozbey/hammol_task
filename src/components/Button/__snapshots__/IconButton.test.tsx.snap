// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing IconButton Snapshots 1`] = `
<DocumentFragment>
  <button
    class="sc-gsnTZi chukKY"
  >
    <span
      class="material-icons"
    >
      more_vert
    </span>
  </button>
</DocumentFragment>
`;
